src/gluonnlp/data/transforms.py

```diff
@@ -1011,6 +1057,84 @@ def convert_tokens_to_ids(self, tokens):
         return self.vocab.to_indices(tokens)
 
 
+class BERTSPTokenizer(BERTTokenizer):
+    r"""End-to-end SentencePiece tokenization for BERT models.
+    It works best with BERTSentenceTransform()
+
+    Parameters
+    ----------
+    path : str
+        Path to the pre-trained subword tokenization model.
+    vocab : gluonnlp.Vocab
+        Vocabulary for the corpus.
+    num_best : int, default 0
+        A scalar for sampling subwords. If num_best = {0,1}, no sampling is performed.
+        If num_best > 1, then samples from the num_best results.
+        If num_best < 0, then assume that num_best is infinite and
+        samples from the all hypothesis (lattice) using forward-filtering-and-backward-sampling
+        algorithm.
+    alpha : float
+        A scalar for a smoothing parameter. Inverse temperature for probability rescaling.
+    lower : bool, default True
+        Whether the text strips accents and convert to lower case.
+        If you use the BERT pre-training model,
+        lower is set to False when using the cased model,
+        otherwise it is set to True.
+    max_input_chars_per_word : int, default 200
+
+    Examples
+    --------
+    >>> url = 'http://repo.mxnet.io/gluon/dataset/vocab/test-682b5d15.bpe'
+    >>> f = gluon.utils.download(url, overwrite=True)
+    -etc-
+    >>> bert_vocab = gluonnlp.vocab.BERTVocab.from_sentencepiece(f)
+    >>> sp_tokenizer = BERTSPTokenizer(f, bert_vocab, lower=True)
+    >>> sentence = 'Better is to bow than break.'
+    >>> sp_tokenizer(sentence)
+    ['▁better', '▁is', '▁to', '▁b', 'ow', '▁than', '▁brea', 'k', '▁', '.']
+    """
+
+    def __init__(self,
+                 path,
+                 vocab,
+                 num_best=0,
+                 alpha=1.0,
+                 lower=True,
+                 max_input_chars_per_word=200):
+        super(BERTSPTokenizer, self).__init__(vocab, lower,
+                                              max_input_chars_per_word)
+        self._path = path
+        self._num_best = num_best
+        self._alpha = alpha
+        self.sentencepiece = None
+
+    def _activate_sp(self):
+        self.sentencepiece = SentencepieceTokenizer(self._path, self._num_best,
+                                                    self._alpha)
```

Comment:
any reason not to initialize this in constructor?

Reply:
I think the comment below explains it: 
```
#Swig object can not be pickled when multiprocessing.
```